### ISSUE TITLE
Update DID

### ISSUE THREAD

**Issue by vongohren** [decentralized-identity/ion-tools#3]
It is not clear to me how you do an update on the DID doc.
Is it basically just new ION.DID, with same did doc, but with some changes to f.ex service endpoint?

Or any other way?

**Comment by csuwildcat** [comment]
Updates are supported in ION the network, but the SDK/lib doesn't have those op generation routines yet - we are working to add these now.

**Comment by thehenrytsai** [comment]
@vongohren, while waiting on support for DID updates at the `ion-tools` and `ion-sdk` level which should be coming in the next couple of weeks, the [Sidetree REST API](https://identity.foundation/sidetree/api/) already supports all the available operations (create/update/recover/deactivate).

**Comment by vongohren** [comment]
Great! Thanks for the update @csuwildcat @thehenrytsai!

**Comment by vongohren** [comment]
How is this going along?

**Comment by codi0** [comment]
@csuwildcat - I noticed the [pull request in ion-sdk](https://github.com/decentralized-identity/ion-sdk/pull/14) for update/recovery/deactivate is merged into the main branch there, but without a new version deployed. Is it waiting on further changes/testing before that can happen?

Anything we can do to help get it over the line?

**Comment by vongohren** [comment]
I would love to try this out next week, when do I know its updated?

**Comment by csuwildcat** [comment]
@vongohren @codi0  I almost have this done - code is ready, just testing it and doing docs for release. I'll have it by end of week, promise!

**Comment by codi0** [comment]
> @vongohren @codi0 I almost have this done - code is ready, just testing it and doing docs for release. I'll have it by end of week, promise!

Awesome :)

**Comment by csuwildcat** [comment]
Hey folks, just pushed out a build with support for all the rest of the DID operations: https://github.com/decentralized-identity/ion-tools#generateoperationtype-contents-commit-async

**Comment by Ajmasta** [comment]
Hello,

Sorry to comment in an old thread but I am having trouble understanding how to update an already existing DID.
The docs always show that you re-create a new DID before doing operations such as updating or recovering.
However, I don't understand how to fetch an already existing DID and work on it. I have some experience with DID:ETHR, where you can put the private key in a request while fetching a DID, but I failed to do the same here.

Could anyone guide me on how to update an already existing DID?

thank you
